### PR TITLE
Add deprecated api version policy

### PR DESCRIPTION
**Pull request by yoavrotems**

### CODE REVIEW

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"apps/v1beta2": {
+		"Deployment": "apps/v1",
+		"DaemonSet": "apps/v1",
+		"StatefulSet": "apps/v1",
+	},
+}
+
+# Get all containers which use deprecated api versions
+getDeprecatedApi[name] {
+	allContainers := kubernetes.containers[_]
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:35]: containers is not relevant in this policy. some resources doesn't have containers

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	},
+	"apps/v1beta2": {
+		"Deployment": "apps/v1",
+		"DaemonSet": "apps/v1",
+		"StatefulSet": "apps/v1",
+	},
+}
+
+# Get all containers which use deprecated api versions
+getDeprecatedApi[name] {
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:34]: style comment: rules shouldn't start with "get"

**Review by itaysk — CHANGES_REQUESTED** [review]

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package appshield.kubernetes.KSV101
+
+import data.lib.kubernetes
+import data.lib.utils
+
+__rego_metadata__ := {
+	"id": "KSV101",
+	"title": "Using A Deprecated API Version",
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:8]:
```suggestion
	"title": "Resource is using A deprecated API version",
```

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	"version": "v1.0.0",
+	"severity": "Critical",
+	"type": "Kubernetes Security Check",
+	"description": "Check if any objects are using a deprecated version of API.",
+	"recommended_actions": "Don't use deprecated API versions",
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:13]:
```suggestion
	"recommended_actions": "Update to use a newer API version.",
```

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package appshield.kubernetes.KSV101
+
+import data.lib.kubernetes
+import data.lib.utils
+
+__rego_metadata__ := {
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:6]: missing __rego_input__

**Review by itaysk — COMMENTED** [review]
metadata comments

File: kubernetes/policies/general/deprecated_api_version.rego [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	"id": "KSV101",
+	"title": "Using A Deprecated API Version",
+	"version": "v1.0.0",
+	"severity": "Critical",
```
Comment (itaysk) [inline kubernetes/policies/general/deprecated_api_version.rego:10]: I don't think this is a Critical severity policy

**Review by itaysk — COMMENTED** [review]
General comment, I find this policy a bit odd because it's highly dependent on the kubernetes version that the customer is intending to deploy the resource at, and we don't have this information. for example, we might recommend to update to apps/v1, but this version is unsupported on the customer's cluster. If we do intend to keep this policy, we should thoroughly include all the version changes that are listed here: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#removed-apis-by-release

**Review by itaysk — APPROVED** [review]
LGTM, consider changing the severity to low

**Comment by knqyf263** [comment]
Thanks. I'll take a look at last.

**Pull request merged**